ddl/index.go

```diff
@@ -161,35 +162,39 @@ func checkIndexColumn(col *model.ColumnInfo, ic *ast.IndexColName) error {
 }
 
 func getIndexColumnLength(col *model.ColumnInfo, colLen int) (int, error) {
-	// Take care of the sum of length of all index columns.
+	length := types.UnspecifiedLength
```

Comment:
Please add some comments for this function, and point out the effect of the charset.